### ISSUE TITLE
Prevent comparison of null with integer

### ISSUE THREAD

**Issue by bdsl** [vimeo/psalm#6869]
With code like https://psalm.dev/r/890c56e8d0 (or a more complicated equivalent) I think it's often easy to forget to consider the case where the variable is null.

It might be useful for Psalm to emit an issue to stop comparisons with nullable values, and force the programmer to handle the null explicitly as a special case.

**Comment by brunopinna** [comment]
I agree. Comparisons with nullable values are bug triggers. It would be definitely useful. 

Thank you very much @bdsl for highlighting this!

**Comment by orklah** [comment]
We'll gladly accept a PR adding this as a opt-in config in the xml :)

**Comment by bdsl** [comment]
Thanks @orklah, I might make the PR. But if anyone else wants to do it don't wait for me, I'll come back and comment here before I spend lots of time on it.

**Comment by bdsl** [comment]
I suppose this config option would work similar to e.g. [findUnusedCode](https://psalm.dev/docs/running_psalm/configuration/#findunusedcode). But I wonder if there's a more general solution for config where a whole set of issues could be set to suppress by default, and then for each one the config could override to make it output as either info or error according to preference.

That might make it easier to introduce new issues, initially as suppressed by default and then perhaps becoming errors in new major version or at least after more thorough testing and consideration.

**Comment by orklah** [comment]
This might makes sense just looking at the config but unused code (and taint analysis) use a whole module that adds operations all along the analysis. It would not makes sense to do those no matter what and then suppress the issues afterward

But yeah, if we want to create a new level for Psalm one day, we'll probably revisit configs that adds strictness to put in there
